internal/controller/cephfscg/cghandler.go

```diff
@@ -0,0 +1,442 @@
+package cephfscg
+
+import (
+	"context"
+	"fmt"
+
+	volsyncv1alpha1 "github.com/backube/volsync/api/v1alpha1"
+	"github.com/go-logr/logr"
+	snapv1 "github.com/kubernetes-csi/external-snapshotter/client/v7/apis/volumesnapshot/v1"
+	ramendrv1alpha1 "github.com/ramendr/ramen/api/v1alpha1"
+	"github.com/ramendr/ramen/internal/controller/util"
+	"github.com/ramendr/ramen/internal/controller/volsync"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func NewVSCGHandler(
+	ctx context.Context,
+	k8sClient client.Client,
+	instance *ramendrv1alpha1.VolumeReplicationGroup,
+	volumeGroupSnapshotSource *metav1.LabelSelector,
+	vsHandler *volsync.VSHandler,
+	cgName string,
+	logger logr.Logger,
+) VSCGHandler {
+	log := logger.WithName("VSCGHandler")
+
+	cgHandler := &cgHandler{
+		ctx:                       ctx,
+		Client:                    k8sClient,
+		instance:                  instance,
+		VSHandler:                 vsHandler,
+		volumeGroupSnapshotSource: volumeGroupSnapshotSource,
+		cgName:                    cgName,
+		logger:                    log,
+	}
+
+	if instance.Spec.Async != nil {
```

Comment:
Can we set a default value if the instance.Spec.Async is nil?  like
```
	if instance.Spec.Async != nil {
		log.Info("Set Async config")

		cgHandler.volumeSnapshotClassSelector = instance.Spec.Async.VolumeSnapshotClassSelector
		cgHandler.ramenSchedulingInterval = instance.Spec.Async.SchedulingInterval
		cgHandler.volumeGroupSnapshotClassSelector = instance.Spec.Async.VolumeGroupSnapshotClassSelector
	} else {
		cgHandler.ramenSchedulingInterval = <DefaultSchedulingInterval>
	}
```


I am concerned that if Async is not assigned(might be a time) and we remove this check, it might lead to error of invalid memory address or nil pointer dereference. the `NewVSHandler` also has this check: https://github.com/RamenDR/ramen/blob/97f1457b8fc1a3a367845effdbae6404988bec52/internal/controller/volsync/vshandler.go#L93

Reply:
So Async is set on RDR but not on MDR. If it is not set, you should avoid running the default, as this indicates that we are running a MetroDR, and no consistency group code should be executed. If you get an invalid memory address, then that's a good thing :)

Reply:
removed the Async check